### PR TITLE
[11.0][MIG] module_analysis: Backport to 11.0

### PR DESCRIPTION
**Pull request by chusamo**

### CODE REVIEW

**Comment by legalsylvain** [comment]
Hi chusamo. Thanks for porting this module. Could you respect commit history, cherry picking commits ?
Thanks.

**Comment by misern2** [comment]
@chusamo please follow the guide [https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-11.0](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-11.0)

Thanks

**Comment by chusamo** [comment]
> Hi chusamo. Thanks for porting this module. Could you respect commit history, cherry picking commits ?
> Thanks.
@legalsylvain Done. Should Travis install the python library pygount or is all right with the build failed?

**Review by anasuarez1 — APPROVED** [review]
Functional review OK

**Review by Jordijane98 — APPROVED** [review]
Looks great for me!

**Comment by chusamo** [comment]
@legalsylvain Could you check this and merge if you think is good?
Thanks

**Comment by legalsylvain** [comment]
Hi @chusamo. 
Thanks for your work
1) could you check why travis is red, and fix the problem ? 
2) I'm not maintainers of this repo. you should ask to server-tools mainteiners to merge it.

thanks !

**Comment by chusamo** [comment]
> Hi @chusamo.
> Thanks for your work
> 
> 1. could you check why travis is red, and fix the problem ?
> 2. I'm not maintainers of this repo. you should ask to server-tools mainteiners to merge it.
> 
> thanks !

I thought you can merge, sorry.  Do you know if travis should found external dependences on v11? [IMAGE](https://ibb.co/JjpCn8W)

**Comment by chusamo** [comment]
Can anyone merge this PR? runbot is not finding the external dependence pygount https://ibb.co/JjpCn8W

**Comment by chusamo** [comment]
@LoisRForgeFlow @dreispt @simahawk Hello Could you check and merge this PR please?

**Comment by simahawk** [comment]
@chusamo if the build is red we don't merge. If `pygount` is missing, just add it to https://github.com/OCA/server-tools/blob/11.0/requirements.txt (it has been done in v13 for instance).

**Comment by chusamo** [comment]
> @chusamo if the build is red we don't merge. If `pygount` is missing, just add it to https://github.com/OCA/server-tools/blob/11.0/requirements.txt (it has been done in v13 for instance).

Thank you simahawk I added in #2037

**Comment by misern2** [comment]
@chusamo You don't need to create a new PR. Just create add the file to the current PR and then squash all commits.

**Comment by chusamo** [comment]
> @chusamo You don't need to create a new PR. Just create add the file to the current PR and then squash all commits.

Oh I see, thank you!

**Comment by misern2** [comment]
@chusamo Travis is red. Can you check it ?

File: module_analysis/models/ir_module_module.py
Comment (renatonlima) [inline module_analysis/models/ir_module_module.py]: This method call is deprecation, it was fixed in 12.0 branch, can you cherry-pick these commits? 

https://github.com/OCA/server-tools/pull/2036/commits/755569b57ba595f555db289a2fbc9b1e3cb358cf
https://github.com/OCA/server-tools/pull/2038/commits/315f7e8f22ee6f1ef88b423bfc451b900471df34

**Review by renatonlima — CHANGES_REQUESTED** [review]

File: module_analysis/models/ir_module_module.py
Comment (chusamo) [reply module_analysis/models/ir_module_module.py]: Done Thank you for your feedback :)

**Comment by renatonlima** [comment]
Hi @chusamo 


Can you check travis error? https://travis-ci.com/github/OCA/server-tools/jobs/490307054#L345

**Comment by chusamo** [comment]
> Hi @chusamo
> 
> Can you check travis error? https://travis-ci.com/github/OCA/server-tools/jobs/490307054#L345

@misern2  @renatonlima  Waiting for this fix #2041  because it fails when we analyse the code. [image](https://ibb.co/TtR1TtR)

**Comment by thomaspaulb** [comment]
@chusamo Please fix conflicts

**Pull request closed without merging**
